tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g expected_output
+
+function self_update_test_helper()
+{
+  local update_type="$1"
+  local kw_branch="${update_type}"
+  local base_version
+  local distro
+  local container
+  local output
+  local expected_output
+  local head_commit_hash
+
+  # Get the short commit hash of the branch
+  head_commit_hash=$(git rev-parse --short origin/"${kw_branch}")
+
+  # Get the base version from the specified branch
+  base_version=$(git show "${kw_branch}":./src/VERSION | head --lines 1)
```

Comment:
Like in the git command above, we assume some things about the dev's local kw repo. For example, I didn't have the `master` branch update with the upstream, and `src/VERSION` still contains `alpha` as the base version in this local branch. This causes the expected base version to be wrong (see screenshot below).

![Screenshot from 2024-03-10 19-23-35](https://github.com/kworkflow/kworkflow/assets/95055784/a31402f9-1f19-48da-8ffb-1b1300b13415)

**Note:** As in the previous screenshot, the output differs because I cleaned the extra output and used `assert_equals_helper`. Also, I "fixed" the problem pointed in the screenshot to make sure that the test failed strictly due to the expected base version being wrong.